pkg/scraper/scraper.go

```diff
@@ -141,17 +143,21 @@ func (c *scraper) Scrape(baseCtx context.Context) *storage.MetricsBatch {
 			responseChannel <- m
 		}(node)
 	}
-
-	res := &storage.MetricsBatch{}
-
+	res := &storage.MetricsBatch{
+		Nodes: make(map[string]storage.NodeMetricsPoint),
+		Pods:  make(map[apitypes.NamespacedName]storage.PodMetricsPoint),
+	}
 	for range nodes {
 		srcBatch := <-responseChannel
 		if srcBatch == nil {
 			continue
 		}
-
-		res.Nodes = append(res.Nodes, srcBatch.Nodes...)
-		res.Pods = append(res.Pods, srcBatch.Pods...)
+		for nodeName, nodeMetricsPoint := range srcBatch.Nodes {
```

Comment:
Previously storage was responsible for finding duplicates, now it should be done here.